rest/adminapitest/admin_api_test.go

```diff
@@ -2141,6 +2141,73 @@ func TestHandleCreateDB(t *testing.T) {
 	rest.RequireStatus(t, resp, http.StatusBadRequest)
 }
 
+func TestHandleCreateDBJsonName(t *testing.T) {
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("This test only works against Couchbase Server")
+	}
+	testCases := []struct {
+		name        string
+		JSONname    string
+		expectError bool
+	}{
+		{
+			name:        "Name match",
+			JSONname:    "db1",
+			expectError: false,
+		},
+		{
+			name:        "Name mismatch",
+			JSONname:    "dummy",
+			expectError: true,
+		},
+		{
+			name:        "No JSON Name",
+			JSONname:    "",
+			expectError: false,
+		},
+	}
+
+	for _, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
+			tb := base.GetTestBucket(t)
+			defer tb.Close()
+
+			rt := rest.NewRestTester(t, &rest.RestTesterConfig{
+				CustomTestBucket: tb,
+				PersistentConfig: true,
+				MutateStartupConfig: func(config *rest.StartupConfig) {
+					config.Bootstrap.ConfigUpdateFrequency = base.NewConfigDuration(0)
+				},
+			})
+			defer rt.Close()
+
+			var resp *rest.TestResponse
+			if test.name != "No JSON Name" {
+				resp = rt.SendAdminRequest(http.MethodPut, "/db1/",
+					fmt.Sprintf(
+						`{"bucket": "%s", "name": "%s", "num_index_replicas": 0, "enable_shared_bucket_access": %t, "use_views": %t}`,
```

Comment:
Instead of having nearly the same code here as L2195, a variable could be set here like`DbConfigJson = ``"name":"`` + test.JSONname + ``"``, `which can then be inserted in to the database config on L2195. 

This way, if the variable is not blank, it will be inserted in to the database config, otherwise it will not change the database config.

Reply:
Added in commit https://github.com/couchbase/sync_gateway/pull/5830/commits/a3d2ce57cdc84f9eab786c4923e28cc0a0add5a5 
As well as above requested changes